src/main/java/org/spongepowered/api/util/CatalogBuilder.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.util;
+
+import org.spongepowered.api.CatalogType;
+import org.spongepowered.api.plugin.PluginContainer;
+import org.spongepowered.api.text.translation.Translation;
+
+/**
+ * A base builder to construct {@link CatalogType}s.
+ */
+public interface CatalogBuilder<C extends CatalogType, B extends ResettableBuilder<C, B>> extends ResettableBuilder<C, B> {
+
+    /**
+     * Sets the name of the {@link CatalogType}. Defaults to {@link #id(String)}.
+     *
+     * @param name The name
+     * @return This builder for chaining
+     */
+    B name(String name);
+
+    /**
+     * Sets the name of the {@link CatalogType} as a {@link Translation}. Defaults to {@link #id}.
+     *
+     * @param translation The name translation
+     * @return This builder for chaining
+     */
+    B name(Translation translation);
```

Comment:
I'm not sure on the name here. Because on the server the name is something static usually english. However the translation is something dynamic. Maybe rename it to translation andexplicitly state that the english content will be used for the name as well.

Reply:
Some `CatalogType`s are `Translatable`, this would also apply this translation, and the plain version would be used as `getName` value.

Reply:
Thats why `translation` might be an alternative.

Reply:
`Translarion extends Text`, iirc, why just not `Text`?

Reply:
`Translation` doesn't extend `Text`, but `TranslatableText` does.